### PR TITLE
searchsorted overrides, resolves #85

### LINKED ISSUE #85: `searchsorted` causes segfault
```julia

julia> using OffsetArrays

julia> o = OffsetArray([2,2,3],-1:1)
3-element OffsetArray(::Array{Int64,1}, -1:1) with eltype Int64 with indices -1:1:
 
2
 2
 3

julia> searchsorted(o,1)
Segmentation fault: 11
```
Using the debugger reveals its a strange bug:
```julia
julia> @enter searchsorted(o,1)
In #searchsorted#6(lt, by, rev, order, , v, x) at sort.jl:296
>1  1 ─ %1 = (Base.Order.ord)(lt, by, rev, order)
 2  │   %2 = (searchsorted)(v, x, %1)
 3  └──      return %2

About to run: (Base.Order.ord)(isless, identity, nothing, Base.Order.ForwardOrdering())
1|debug> n
ERROR: BoundsError: attempt to access 3-element OffsetArray(::Array{Int64,1}, -1:1) with eltype Int64 with indices -1:1 at index [9223372036854775807]
Stacktrace:
 [1] throw_boundserror(::OffsetArray{Int64,1,Array{Int64,1}}, ::Tuple{Int64}) at abstractarray.jl:538
 [2] checkbounds(::OffsetArray{Int64,1,Array{Int64,1}}, ::Tuple{Int64}) at abstractarray.jl:503
 [3] getindex(::OffsetArray{Int64,1,Array{Int64,1}}, ::Int64) at /Users/solver/.julia/packages/OffsetArrays/vIbpP/src/OffsetArrays.jl:135
 [4] searchsorted(::OffsetArray{Int64,1,Array{Int64,1}}, ::Int64, ::Int64, ::Int64, ::Base.Order.ForwardOrdering) at sort.jl:214
 [5] searchsorted(::OffsetArray{Int64,1,Array{Int64,1}}, ::Int64, ::Base.Order.ForwardOrdering) at sort.jl:294
 [6] #searchsorted#6(::typeof(isless), ::typeof(identity), ::Nothing, ::Base.Order.ForwardOrdering, ::typeof(searchsorted), ::OffsetArray{Int64,1,Array{Int64,1}}, ::Int64) at sort.jl:296
 [7] searchsorted(::OffsetArray{Int64,1,Array{Int64,1}}, ::Int64) at sort.jl:296
```

PS I want this to support offset `BlockArray`s: https://github.com/JuliaArrays/BlockArrays.jl/pull/95

### PR DESCRIPTION
**Pull request by dlfivefifty**
This works around a segfault in Base's `searchsorted` with offset indices, see #85 and https://github.com/JuliaLang/julia/issues/33977. 

This should eventually be back-ported to Base: perhaps with a simple check in `Base.searchsorted`:
```julia
lo < 0 && return _safe_searchsorted(...)
```

### CODE REVIEW

File: src/OffsetArrays.jl [hunk 1]
```diff
@@ -278,4 +278,59 @@ end
 
 no_offset_view(A::OffsetArray) = no_offset_view(parent(A))
 
+
+####
+# work around for segfault in searchsorted*
+#  https://github.com/JuliaLang/julia/issues/33977
+####
+
+function Base.searchsorted(v::AbstractVector, x, ilo::T, ihi::T, o::Base.Ordering) where T<:Integer
```
Comment (fredrikekre) [inline src/OffsetArrays.jl]: Why is this method needed?

Comment (dlfivefifty) [reply src/OffsetArrays.jl]: Sorry that was a mistake

File: src/OffsetArrays.jl [hunk 2]
```diff
@@ -283,54 +283,103 @@ no_offset_view(A::OffsetArray) = no_offset_view(parent(A))
 # work around for segfault in searchsorted*
 #  https://github.com/JuliaLang/julia/issues/33977
 ####
-
+if VERSION ≤ v"1.2"
+	# ambiguity warnings in earlier versions
+	function Base.searchsorted(v::OffsetArray, x, ilo::Int, ihi::Int, o::Base.Ordering)
```
Comment (fredrikekre) [inline src/OffsetArrays.jl]: Can you move the bodies out to private methods instead of duplicating them?

**Comment by dlfivefifty** [comment]
The test errors on nightly seem unrelated:
```
ERROR: Error when installing package CatIndices:
144MethodError: no method matching install_archive(::Array{String,1}, ::Base.SHA1, ::String)
145Closest candidates are:
146  install_archive(!Matched::Array{Pair{String,Bool},1}, ::Base.SHA1, ::String) at /
```

**Comment by fredrikekre** [comment]
https://github.com/JuliaLang/Pkg.jl/issues/1514

File: .travis.yml [hunk 3]
```diff
@@ -8,6 +8,8 @@ julia:
     - 0.7
     - 1.0
     - 1.1
+    - 1.2
+    - 1.3
     - nightly
```
Comment (fredrikekre) [inline .travis.yml]: Maybe drop 0.7, 1.1 and 1.2 since they are [unmaintained](https://julialang.org/downloads/oldreleases.html)?

**Comment by fredrikekre** [comment]
If you also bump the patch number we can release directly.

**Pull request merged**